File: src/test/java/ulovdomov/pages/LoginPage.java
package ulovdomov.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;

public class LoginPage extends BasePage {
    private static final By HAMBURGER_MENU   = By.cssSelector("[data-test='navbar.hamburgerButton']");
    private static final By LOGIN_LINK       = By.xpath("//span[text()='Přihlásit se']");
    private static final By EMAIL_INPUT      = By.cssSelector("[data-test='loginModal.signIn.form.emailInput']");
    private static final By PASSWORD_INPUT   = By.cssSelector("[data-test='loginModal.signIn.form.passwordInput']");
    private static final By SUBMIT_BUTTON    = By.cssSelector("[data-test='loginModal.signIn.form.button']");

    public LoginPage(WebDriver driver) {
        super(driver);
    }

    public void openLoginModal() {
        waitForClickable(HAMBURGER_MENU).click();
        waitForClickable(LOGIN_LINK).click();
        waitForVisible(EMAIL_INPUT);
    }

    public void login(String email, String password) {
        WebElement emailField = waitForClickable(EMAIL_INPUT);
        emailField.clear();
        emailField.sendKeys(email);

        WebElement passwordField = waitForClickable(PASSWORD_INPUT);
        passwordField.clear();
        passwordField.sendKeys(password);

        waitForClickable(SUBMIT_BUTTON).click();
    }

    public boolean isLoggedIn() {
        try {
            wait.until(ExpectedConditions.invisibilityOfElementLocated(EMAIL_INPUT));
            return true;
        } catch (Exception e) {
            return false;
        }
    }
}

File: pages/login.page.ts
import { type Page, type Locator, expect } from '@playwright/test';
import { BasePage } from './base.page';

export class LoginPage extends BasePage {
  readonly menuButton: Locator;
  readonly loginMenuLink: Locator;
  readonly emailInput: Locator;
  readonly passwordInput: Locator;
  readonly submitButton: Locator;
  readonly closeModalButton: Locator;

  constructor(page: Page) {
    super(page);
    this.menuButton = page.getByRole('button', { name: /otevřít menu/i });
    this.loginMenuLink = page.getByText('Přihlásit se').first();
    this.emailInput = page.locator('[data-test="loginModal.signIn.form.emailInput"]');
    this.passwordInput = page.locator('[data-test="loginModal.signIn.form.passwordInput"]');
    this.submitButton = page.locator('[data-test="loginModal.signIn.form.button"]');
    this.closeModalButton = page.locator('[data-test="loginModal.closeButton"]');
  }

  /** Open the login modal from any page */
  async openLoginModal() {
    await this.menuButton.waitFor({ state: 'visible' });
    await this.menuButton.click();
    await this.loginMenuLink.waitFor({ state: 'visible' });
    await this.loginMenuLink.click();
    await this.emailInput.waitFor({ state: 'visible' });
  }

  /** Fill and submit login form */
  async login(email: string, password: string) {
    await this.emailInput.fill(email);
    await expect(this.emailInput).toHaveValue(email);
    await this.passwordInput.fill(password);
    await expect(this.passwordInput).toHaveValue(password);
    await this.submitButton.click();
    await this.waitForPageLoad();
  }

  /** Verify login was successful by checking modal disappeared */
  async verifyLoggedIn() {
    await expect(this.emailInput).toBeHidden({ timeout: 10000 });
  }
}
